vision_evaluation/evaluators.py

```diff
@@ -686,7 +686,37 @@ def _summarize(eval_result, ap=1, iouThr=None, areaRng='all', maxDets=300, catId
             mean_s = np.mean(s[s > -1])
         return mean_s
 
+    def remap_cat_ids(self):
+        """Map target category ids to continuous ones to avoid the index overflow issue during coco mAP computation.
+           Then any predicted category ids not shown in targets will be appended. e.g. targets:[0,2,4], predictions:[0,1,2,3],
+           the category ids are remapped as 0->0, 2->1, 4->2, 1(not shown in targets)->3, 3(not shown in targets)->4"""
+
+        # Create the map.
+        target_cat_ids = np.unique([bbox[0] for bboxes in self.targets for bbox in bboxes])
+        predict_cat_ids = np.unique([bbox[0] for bboxes in self.predictions for bbox in bboxes])
+        cate_id_old_to_new = {cate_id: idx for idx, cate_id in enumerate(target_cat_ids)}
+
+        for cate_id in predict_cat_ids:
+            if cate_id not in cate_id_old_to_new:
+                cate_id_old_to_new[cate_id] = len(cate_id_old_to_new)
+
+        # Apply the map.
+        def _apply_cate_id_map(gt_or_pred):
+            for bboxes in gt_or_pred:
+                for bbox in bboxes:
+                    bbox[0] = cate_id_old_to_new[bbox[0]]
+
+        _apply_cate_id_map(self.targets)
+        _apply_cate_id_map(self.predictions)
+
+        # Return the new id to old id map.
+        cate_id_new_to_old = [None] * len(cate_id_old_to_new)
```

Comment:
It might be easier if we create this list first.
```
target_cat_ids = set([b[0] for bboxes in self.targets for b in bboxes])
predict_cat_ids = set([b[0] for bboxes in self.predictions for b in bboxes])
cate_id_new_to_old = sorted(list(target_cat_ids)) + sorted(list(predict_cat_ids - target_cat_ids))
cate_id_old_to_new = {o: n for n, o in enumerate(cate_id_new_to_old)}
```

Reply:
Thanks, updated!